server/middleware.go

```diff
@@ -33,6 +33,7 @@ func withMetrics(
 		// we assume that every route will set the status header
 		versionByte, err := parseVersionByte(w, r)
 		if err != nil {
+			recordDur(w.Header().Get("status"), string(mode), string(versionByte))
```

Comment:
This w.Header().Get("status") thing is buggy, I have an issue about this somewhere. But let's keep it like this for now and fix it in another PR.